drivers/ethernet/eth.h

```diff
@@ -42,4 +47,37 @@ static inline void gen_random_mac(uint8_t *mac_addr, uint8_t b0, uint8_t b1, uin
 	sys_rand_get(&mac_addr[3], 3U);
 }
 
+static inline int net_eth_mac_load(const struct net_eth_mac_config *cfg, uint8_t *mac_addr)
+{
+	if (cfg == NULL || cfg->type == NET_ETH_MAC_DEFAULT) {
+		return -ENODATA;
+	}
+
+	/* Copy the static part */
```

Comment:
`__ASSERT_NO_MSG(cfg->addr_len <= NET_ETH_ADDR_LEN)`?

Reply:
You would already get a compiler error if the amount of bytes is larger with `NET_ETH_MAC_DT_DEFINE`.

EDIT: but it doesn't hurt, I guess

Reply:
Right, indeed. Discard my comment.